src/components/reusables/table/headerFactory.ts

```diff
@@ -0,0 +1,21 @@
+import { Column, FormatterParams } from "./Table";
+
+import { ITooltipParams } from "ag-grid-community";
+
+export const createHeader = <T>(
+    headerName: string,
+    field: string,
+    valueFormatter?: (params: FormatterParams<T>) => string,
+    width?: number,
+    tooltip?: (params: ITooltipParams) => string,

```

Comment:
put this as a defaulted parameter then you don't need to check for it being defined in the function body.
```suggestion
    tooltip: (params: ITooltipParams) => string = params => params.value,

```